File: LeetCode/3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums) {
        int ans = 0;
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int s = max(0, i - nums[i]);
            for (int j = s; j <= i; j++) {
                ans += nums[j];
            }
        }
        return ans;
    }
};

File: LeetCode/3427-sum-of-variable-length-subarrays/3427-sum-of-variable-length-subarrays.py
class Solution:
    def subarraySum(self, nums: List[int]) -> int:
        ans = 0
        n = len(nums)
        for i in range(n):
            s = max(0, i - nums[i])
            for j in range(s, i + 1):
                ans += nums[j]
        return ans